geonode_mapstore_client/client/js/plugins/Legend.jsx

```diff
@@ -31,10 +35,16 @@ function Legend({
         </div>
         <ul className="gn-legend-list" style={{display: expandLegend ? 'inline-block' : 'none'}}>
             {layers.map((layer, ind) => <Fragment key={ind}>
-                {!hideLayerTitle &&
-                    <li className="gn-legend-list-item"><p>{layer.title}</p></li>
-                }
-                <li>
+                <li className="gn-legend-list-item"><VisibilityCheck key="visibilitycheck"
+                    tooltip={layer.loadingError === 'Warning' ? 'toc.toggleLayerVisibilityWarning' : 'toc.toggleLayerVisibility'}
+                    node={layer}
+                    propertiesChangeHandler={propertiesChangeHandler} /><p>{layer.name || layer.title}</p></li>
```

Comment:
@DavidQuartz you can use onUpdateNode(layer.id, 'layers', { visibility: ... }) also for this so we avoid to use two different callback for the same action